internal/factsengine/gatherers/corosynccmapctl.go

```diff
@@ -0,0 +1,47 @@
+package gatherers
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/internal/utils"
+)
+
+const (
+	CorosyncCmapCtlFactKey = "corosync-cmapctl"
+)
+
+type CorosyncCmapctlGatherer struct {
+	executor CommandExecutor
+}
+
+func NewCorosyncCmapctlGatherer() *CorosyncCmapctlGatherer {
+	return &CorosyncCmapctlGatherer{
+		executor: Executor{},
+	}
+}
+
+func (s *CorosyncCmapctlGatherer) Gather(factsRequests []FactRequest) ([]Fact, error) {
+	facts := []Fact{}
+	log.Infof("Starting %s facts gathering process", CorosyncCmapCtlFactKey)
+
+	corosyncCmapctl, err := s.executor.Exec(
+		"corosync-cmapctl", "-b")
+	if err != nil {
+		return facts, err
+	}
+
+	corosyncCmapctlMap := utils.FindMatches(`(?m)^(\S*)\s\(\S*\)\s=\s(.*)$`, corosyncCmapctl)
+
+	for _, factReq := range factsRequests {
+		if value, ok := corosyncCmapctlMap[factReq.Argument]; ok {
+			fact := NewFactWithRequest(factReq, fmt.Sprint(value))
+			facts = append(facts, fact)
+		} else {
+			log.Warnf("%s gatherer: requested fact %s not found", CorosyncCmapCtlFactKey, factReq.Argument)
```

Comment:
I have used `log.Errorf` in other gatherers, but I guess it doesn't really matter that much.
Once we know how to populate the failed fact gatherings we will need to come back and finish the job hehe